client/src/components/_common/Button/Button.jsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Button as BootstrapButton } from 'reactstrap';
+import Icon from '../Icon';
+
+import '../../../styles/components/c-button.css';
+
+export const TYPES = [
+    '',
+    'primary',
+    'secondary',
+    'success',
+    'info',
+    'warning',
+    'danger',
+    'link'
+];
+
+export const SIZES = [
+    'short',
+    'medium',
+    'large',
+    'small'
+]
+
+const Button = ({text, iconNameBefore, iconNameAfter, type, size, ...props}) => {
```

Comment:
**🛎  Use of `{...props}`**
Please defend your use of `{...props}`. It's use is to be limited in this code base.

<details><summary>Details</summary>

The use of `{...props}`, when approved, should be explained in a comment, and prefixed with something like `// eslint-disable-blah-blah …` to override ES Lint warnings. See [all such instances](https://github.com/TACC/Core-Portal/search?q=jsx-props-no-spreading).

_There is no ES Lint warning, because the introduction of Vite did not preserve our ES Lint rules.)_

</details>
<details><summary>Reference</summary>

- [ESLint React Plugin docs](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-props-no-spreading.md)
- [a blog article about why to limit prop spreading](https://aurelio.me/blog/Why-I-stopped-spreading-props-react/)

</details>

Reply:
I took that from the Reactstrap button, admittedly a remnant when I had an ambiguous approach to this task - consider it gone!